lib/kubernetes-deploy/restart_task.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+require 'kubernetes-deploy/kubeclient_builder'
+
+module KubernetesDeploy
+  class RestartTask
+    include KubernetesDeploy::KubeclientBuilder
+
+    class DeploymentNotFoundError < FatalDeploymentError
+      def initialize(name, namespace)
+        super("Deployment `#{name}` not found in namespace `#{namespace}`. Aborting the task")
+      end
+    end
+
+    class RestartError < FatalDeploymentError
+      def initialize(deployment_name, response)
+        super("Failed to restart #{deployment_name}. " \
+            "API returned non-200 response code (#{response.code})\n" \
+            "Response:\n#{response.body}")
+      end
+    end
+
+    HTTP_OK_RANGE = 200..299
+
+    def initialize(context:, namespace:, logger: KubernetesDeploy.logger)
+      @context = context
+      @namespace = namespace
+      @logger = logger
+      @v1beta1_kubeclient = build_v1beta1_kubeclient(context)
+    end
+
+    def perform(deployments_names)
+      if deployments_names.empty?
+        raise ArgumentError, "#perform takes at least one deployment to restart"
+      end
+      deployments = fetch_deployments(deployments_names.uniq)
+      deployments.each do |record|
+        response = patch_deployment_with_restart(record)
+        if HTTP_OK_RANGE.cover?(response.code)
+          @logger.info "Triggered #{record.metadata.name.inspect} restart"
+        else
+          raise RestartError.new(record.metadata.name, response)
+        end
+      end
+
+      wait_for_rollout(deployments)
+      deployments.each do |record|
+        @logger.info "Successfully restarted #{record.metadata.name.inspect}"
+      end
+    end
+
+    private
+
+    def wait_for_rollout(deployments)
```

Comment:
@KnVerey had to copypaste majority of the method to handle waiting for `Deployment` rollout